### PR TITLE
Only set User-Agent header on browser.

### LINKED ISSUE #139: Refused to set unsafe header "user-agent"
This isn't technically an issue with this library, however it should be noted that neither Chrome nor Safari currently allow setting the "user-agent" header, both throw `Refused to set unsafe header "user-agent"` errors.

It might be responsible to detect whether or not the browser allows for setting this header to avoid spurious errors getting thrown.

I'm happy to open a PR if there's a chance it will go in.  We'd certainly like that, as currently we're getting this error on every page load.

### PR DESCRIPTION
**Pull request by f2prateek**
Closes #139.

### CODE REVIEW

**Comment by f2prateek** [comment]
Ideally we'd only do this in the browsers that don't support this, but didn't seem like there was a good way to do this yet.

File: index.js [hunk 1]
```diff
@@ -220,16 +220,23 @@ class Analytics {
...
       method: 'POST',
       url: `${this.host}/v1/batch`,
       auth: {
         username: this.writeKey
       },
       data,
-      headers: {
-        'user-agent': `analytics-node/${version}`
-      }
+      headers: headers
```
Comment (vadimdemedes) [inline index.js]: Can just be `headers`, like `data` above.

Comment (stephenmathieson) [reply index.js]: I agree, destructuring this looks a bit nicer

Comment (f2prateek) [reply index.js]: updated.

**Comment by f2prateek** [comment]
thanks for the review @vadimdemedes @stephenmathieson !

File: index.js [hunk 2]
```diff
@@ -220,16 +220,23 @@ class Analytics {
...
     }
 
+    // Don't set the user agent if we're not on a browser. The latest spec allows
+    // the User-Agent header (see https://fetch.spec.whatwg.org/#terminology-headers
+    // and https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/setRequestHeader),
+    // but browsers such as Chrome and Safari have not caught up.
+    const headers = {}
+    if (typeof window === 'undefined') {
+      headers['user-agent'] = `analytics-node/${version}`
+    }
```
Comment (sirianni) [inline index.js:230]: The comment and PR title don't match the implementation.  This implementation seems to set the `user-agent` header only if we're **not** on a browser (`window === undefined` means we're not in a browser).

**Pull request merged**